lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsInputFile.java

```diff
@@ -117,7 +119,10 @@ private FSDataInputStream openFile()
             }
             catch (IOException e) {
                 openFileCallStat.recordException(e);
-                throw e;
+                if (e instanceof FileNotFoundException) {
```

Comment:
With this form you are recording the exception in the stat object.  Is that what you want?  If not, I'd move this to a separate catch clause

Reply:
Yeah, the stat object actually has special handling for `FileNotFoundException`